logging: build fails with `the comparison will always evaluate as 'false' for the address of '_fmt' will never be NULL` error

### Describe the bug

Hello,

I encountered the following issue while working on a RISC‑V SoC‑based project. I believe this is a common problem.
```
...
/home/aanikolaev/Projects/MyProject/maic.c:107:301: error: the comparison will always evaluate as 'false' for the address of '_fmt' will never be NULL [-Werror=address]
  107 |         LOG_WRN("Some text");
      |                                                                                                                                                                                                                                                                                                             ^ 
/home/aanikolaev/Projects/MyProject/maic.c:107:69: note: '_fmt' declared here
  107 |         LOG_WRN("Some text");
      |                                                                     ^   
...
```
The warning did not appear until I enabled the `CONFIG_COMPILER_SAVE_TEMPS=y` option for debugging. In a small project a few extra warnings are tolerable, but in a larger code base with many logging calls they quickly become noise. This is especially problematic when `-Werror` is used, because the build stops on the first warning that is turned into an error.

Therefore, I propose the following minimal change to `log_msg.h` to suppress the spurious warning:
```
--- a/include/zephyr/logging/log_msg.h
+++ b/include/zephyr/logging/log_msg.h
@@ -330,7 +330,7 @@ do { \
        int _plen; \
        uint32_t _options = Z_LOG_MSG_CBPRINTF_FLAGS(_cstr_cnt) | \
                          CBPRINTF_PACKAGE_ADD_RW_STR_POS; \
-       if (GET_ARG_N(1, __VA_ARGS__) == NULL) { \
+       if (is_null_no_warn((void *)GET_ARG_N(1, __VA_ARGS__))) { \
                _plen = 0; \
        } else { \
                CBPRINTF_STATIC_PACKAGE(NULL, 0, _plen, Z_LOG_MSG_ALIGN_OFFSET, _options, \
```

This change eliminates the false‑positive “address will never be NULL” warning while preserving the original semantics of the logging macro.

### Regression

- [ ] This is a regression.

### Steps to reproduce

1. `CONFIG_COMPILER_SAVE_TEMPS=y`
2.  `zephyr_compile_options(-Werror)` (on demand)

### Relevant log output

```shell

```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

--- a/include/zephyr/logging/log_msg.h
+++ b/include/zephyr/logging/log_msg.h
@@ -330,7 +330,7 @@ do { \
 	int _plen; \
 	uint32_t _options = Z_LOG_MSG_CBPRINTF_FLAGS(_cstr_cnt) | \
 			  CBPRINTF_PACKAGE_ADD_RW_STR_POS; \
-	if (GET_ARG_N(1, __VA_ARGS__) == NULL) { \
+	if (is_null_no_warn((void *)GET_ARG_N(1, __VA_ARGS__))) { \
 		_plen = 0; \
 	} else { \
 		CBPRINTF_STATIC_PACKAGE(NULL, 0, _plen, Z_LOG_MSG_ALIGN_OFFSET, _options, \

